blog/2024/August/why-meshtastic-uses-managed-flood-router.mdx

```diff
@@ -0,0 +1,37 @@
+---
+title: Why Meshtastic Uses Managed Flood Routing
+description: "Why Meshtastic Uses Managed Flood Routing"
+slug: why-meshtastic-uses-managed-flood-routing
+authors: [thebentern, GUVWAF]
+tags: [meshtastic, technical]
+date: 2024-08-18T12:00
+hide_table_of_contents: false
+image: "/static/img/blog/route_plot.png"
+---
+
+Designing a low-bandwidth wireless mesh network to run on low-power microprocessors with limited memory is challenging. Arguably the simplest mesh routing protocol is Flood Routing: each radio receiving a packet will rebroadcast this again, up to a certain hop limit. Although Meshtastic is based on this strategy, there are a few subtle, but significant enhancements. Most importantly, before a node rebroadcasts, it waits a short while and listens if anyone else is rebroadcasting already. If so, it won’t rebroadcast again. Therefore, “Managed Flood Routing” would be a better name. For more details on the enhancements, please review our [documentation](https://meshtastic.org/docs/overview/mesh-algo/).
+
+
+{/* truncate */}
+
+Since Flood Routing is not very efficient, we realize that this approach is not perfect. The firmware has a number of measures in place to limit traffic in order not to overwhelm your mesh, but there will be a limit to what the mesh can withstand. In attempts to enhance the efficiency, we have evaluated “smarter” routing protocols at times in the past. However, we have yet to find anything that would consistently outperform the current approach in the use-cases and scenarios Meshtastic is currently being utilized. We’ll go over several reasons for why we believe Managed Flood Routing remains a superior approach for Meshtastic.
+
+## Avoids Route Discovery 
+First and foremost, Managed Flood Routing eliminates the need for route discovery or centralized control. In traditional routing algorithms, devices rely on predefined or dynamic routes to forward messages to their destinations. With Managed Flood Routing, you can start messaging immediately after booting your device. Furthermore, route discovery and maintenance leads to overhead, which easily becomes very significant with a low-bandwidth technique like LoRA. In order to maintain routes, either additional control packets are needed, or metadata has to be added to normal traffic, increasing utilization of precious airtime. In static scenarios, this is not a signifcant issue, but when the topology changes often -as we discuss in the next section- the overhead quickly outweighs the benefit of a smarter routing protocol.
```

Comment:
- LoRA should be LoRa to not be confused with "low-rank adaptation".
- signifcant -> significant